pkgs/development/compilers/binaryen/default.nix

```diff
@@ -1,53 +1,61 @@
 { lib, stdenv, cmake, python3, fetchFromGitHub, emscripten,
   gtest, lit, nodejs, filecheck
 }:
-
-stdenv.mkDerivation rec {
-  pname = "binaryen";
-  version = "118";
-
-  src = fetchFromGitHub {
+let
+  testsuite = fetchFromGitHub {
     owner = "WebAssembly";
-    repo = "binaryen";
-    rev = "version_${version}";
-    hash = "sha256-akMW3S2/qUyLK8F77EtnaXPDXvIMpkGfNB2jOD6hQho=";
-  };
-
-  nativeBuildInputs = [ cmake python3 ];
-
-  preConfigure = ''
-    if [ $doCheck -eq 1 ]; then
-      sed -i '/googletest/d' third_party/CMakeLists.txt
-    else
-      cmakeFlagsArray=($cmakeFlagsArray -DBUILD_TESTS=0)
-    fi
-  '';
-
-  nativeCheckInputs = [ gtest lit nodejs filecheck ];
-  checkPhase = ''
-    LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$PWD/lib python3 ../check.py $tests
-  '';
-
-  tests = [
-    "version" "wasm-opt" "wasm-dis"
-    "crash" "dylink" "ctor-eval"
-    "wasm-metadce" "wasm-reduce" "spec"
-    "lld" "wasm2js" "validator"
-    "example" "unit"
-    # "binaryenjs" "binaryenjs_wasm" # not building this
-    "lit" "gtest"
-  ];
-  doCheck = stdenv.hostPlatform.isLinux;
-
-  meta = with lib; {
-    homepage = "https://github.com/WebAssembly/binaryen";
-    description = "Compiler infrastructure and toolchain library for WebAssembly, in C++";
-    platforms = platforms.all;
-    maintainers = with maintainers; [ asppsa willcohen ];
-    license = licenses.asl20;
-  };
-
-  passthru.tests = {
-    inherit emscripten;
+    repo = "testsuite";
+    rev = "e05365077e13a1d86ffe77acfb1a835b7aa78422";
+    hash = "sha256-yvZ5AZTPUA6nsD3xpFC0VLthiu2CxVto66RTXBXXeJM=";
   };
-}
+in
+  stdenv.mkDerivation rec {
+    pname = "binaryen";
+    version = "119";
+
+    src = fetchFromGitHub {
+      owner = "WebAssembly";
+      repo = "binaryen";
+      rev = "version_${version}";
+      hash = "sha256-JYXtN3CW4qm/nnjGRvv3GxQ0x9O9wHtNYQLqHIYTTOA=";
+    };
```

Comment:
```suggestion
    src = fetchFromGitHub {
      owner = "WebAssembly";
      repo = "binaryen";
      rev = "version_${version}";
      hash = "sha256-JYXtN3CW4qm/nnjGRvv3GxQ0x9O9wHtNYQLqHIYTTOA=";
      fetchSubmodules = true;
    };
```

Reply:
Much cleaner. Brain freeze about submodules, in my head this meant that the final derivation would end up bigger, but it should be irrelevant if it's just about build environment. Updated and testing again.

Reply:
Given a lack of information on my part about why this simpler fix failed, I'm inclined to rework using the method proposed earlier. I'm always open to revise to something that handles the submodules more gracefully, but this has been broken long enough that it'd be great to see this resolved.

Reply:
tried to look into it last week, but didn't get to the root of it. I assume that pulling the submodules also pulls in other dependencies overriding their correctly packaged counterparts with the upstream version that get's confused by nix specialties.
I can also try to run nixpkgs-review later.

Reply:
Thank you!